packages/earn-controller/src/EarnController.ts

```diff
@@ -323,43 +310,34 @@ export class EarnController extends BaseController<
     // temporary array of supported chains
     // TODO: remove this once we export a supported chains list from the sdk
     // from sdk or api to get lending and pooled staking chains
-    this.#supportedPooledStakingChains = [
-      CHAIN_IDS.MAINNET,
-      HOODI_TESTNET_CHAIN_ID_HEX,
-    ];
+    this.#supportedPooledStakingChains = [ChainId.ETHEREUM, ChainId.HOODI];
 
     this.#addTransactionFn = addTransactionFn;
 
-    this.#initializeSDK().catch(console.error);
+    this.#selectedNetworkClientId = selectedNetworkClientId;
+
+    this.#initializeSDK(selectedNetworkClientId).catch(console.error);
     this.refreshPooledStakingData().catch(console.error);
     this.refreshLendingData().catch(console.error);
 
-    const { selectedNetworkClientId } = this.messagingSystem.call(
-      'NetworkController:getState',
-    );
-    this.#selectedNetworkClientId = selectedNetworkClientId;
-
     // Listen for network changes
     this.messagingSystem.subscribe(
-      'NetworkController:stateChange',
+      'NetworkController:networkDidChange',
       (networkControllerState) => {
-        if (
-          networkControllerState.selectedNetworkClientId !==
-          this.#selectedNetworkClientId
-        ) {
-          this.#initializeSDK(
-            networkControllerState.selectedNetworkClientId,
-          ).catch(console.error);
-
-          this.refreshPooledStakingVaultMetadata().catch(console.error);
-          this.refreshPooledStakingVaultDailyApys().catch(console.error);
-          this.refreshPooledStakingVaultApyAverages().catch(console.error);
-          this.refreshPooledStakes().catch(console.error);
-
-          // refresh lending data for all chains
-          this.refreshLendingMarkets().catch(console.error);
-          this.refreshLendingPositions().catch(console.error);
-        }
+        this.#initializeSDK(
+          networkControllerState.selectedNetworkClientId,
+        ).catch(console.error);
+
+        // refresh pooled staking data
+        this.refreshPooledStakingVaultMetadata().catch(console.error);
+        this.refreshPooledStakingVaultDailyApys().catch(console.error);
+        this.refreshPooledStakingVaultApyAverages().catch(console.error);
+        this.refreshPooledStakes().catch(console.error);
+
+        // refresh lending data for all chains
+        this.refreshLendingMarkets().catch(console.error);
+        this.refreshLendingPositions().catch(console.error);
+
         this.#selectedNetworkClientId =
```

Comment:
Nit: Is there a reason why this is being stored last in the handler? If other methods need to use it then it won't be up to date until after those methods are called.

Reply:
No reason. The above methods don't need to use it but that's a great callout - I'll change it to be the first thing updated so that if in future methods rely on it we shouldn't create a bug.